subscription.go

```diff
@@ -103,6 +103,12 @@ func (r *marathonClient) registerSubscription() error {
 	case EventsTransportCallback:
 		return r.registerCallbackSubscription()
 	case EventsTransportSSE:
+		if timeout := r.client.config.HTTPClient.Timeout; timeout > 0 {
```

Comment:
In my understanding only case that is not covered by other timeouts (set by client in this PR) is when marathon writes headers and then block sending rest of the response (or send it very slowly). We can replace client.Timeout completely by setting cancel/ctx on every non-sse request - what versions of Go are supported?